packages/preferences/src/browser/preferences-tree-widget.ts

```diff
@@ -236,36 +289,116 @@ export class PreferencesEditorsContainer extends DockPanel {
+        const newFoldersPreferenceEditorWidget = await this.getFoldersPreferencesEditor(newFolderUri);
+        if (newFoldersPreferenceEditorWidget && // new widget is created
+            // the FolderPreferencesEditor is not available, OR the existing FolderPreferencesEditor is displaying the content of a different file
+            (!this._foldersPreferenceEditorWidget || this._foldersPreferenceEditorWidget.editor.uri.parent.parent.toString() !== newFolderUri)) {
+            this.addWidget(newFoldersPreferenceEditorWidget,
+                { ref: this._foldersPreferenceEditorWidget || this._workspacePreferenceEditorWidget || this._userPreferenceEditorWidget });
+            if (this._foldersPreferenceEditorWidget) {
+                this._foldersPreferenceEditorWidget.close();
+                this._foldersPreferenceEditorWidget.dispose();
+            }
+            this._foldersPreferenceEditorWidget = newFoldersPreferenceEditorWidget;
+            this.onFolderPreferenceEditorUriChangedEmitter.fire(newFoldersPreferenceEditorWidget.editor.uri.toString());
+        }
+    }
+
+    protected async getFoldersPreferencesEditor(folder: string | undefined): Promise<PreferencesEditorWidget | undefined> {
+        if (this.workspaceService.saved) {
+            const settingsUri = await this.getFolderSettingsUri(folder);
+            const foldersPreferences = settingsUri && await this.editorManager.getOrCreateByUri(settingsUri) as PreferencesEditorWidget;
+            if (foldersPreferences) {
+                foldersPreferences.title.label = 'Folder';
+                foldersPreferences.title.icon = 'database-icon medium-yellow file-icon';
+                foldersPreferences.title.caption = `Folder Preferences: ${await this.getPreferenceEditorCaption(settingsUri!)}`;
+                foldersPreferences.title.clickableText = new URI(folder).displayName;
+                foldersPreferences.title.clickableTextTooltip = 'Click to manage preferences in another folder';
+                foldersPreferences.title.clickableTextCallback = async (folderUriStr: string) => {
+                    await foldersPreferences.saveable.save();
+                    await this.refreshFoldersPreferencesEditorWidget(folderUriStr);
+                    this.activatePreferenceEditor(PreferenceScope.Folders);
+                };
+                foldersPreferences.scope = PreferenceScope.Folders;
+            }
+            return foldersPreferences;
+        }
+    }
+
+    private async getFolderSettingsUri(folder: string | undefined): Promise<URI | undefined> {
+        if (folder) {
+            const folderUri = new URI(folder);
+            const settingsUri = folderUri.resolve('.theia').resolve('settings.json').toString();
+            if (!(await this.fileSystem.exists(settingsUri))) {
+                await this.fileSystem.createFile(settingsUri);
+            }
+            return new URI(settingsUri);
+        }
```

Comment:
Creating a second time the URI  "settingsUri",  why not using the one created above and use "toString()" when needed a few lines above

Reply:
if i did not misunderstand this comment, you suggested to use the "settingsUri" in line 371.

i believe the current code is already doing what you suggested. line 371 calls `getFolderSettingsUri()`, and uses the returns value after.  the "settingsUri" is created once only.

Reply:
we talked in person. you are right. i will make the requested change.